src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java

```diff
@@ -0,0 +1,397 @@
+	// the system string can be systemname or metadata
+	// (in case it is metadata, it must contain '=')
+	private long getSystemIdByInfo(final String systemInfo) throws Exception {
+		Assert.notNull(this.systems, "The ist of possible systems is empty!");
+		for (SystemResponseDTO system : systems) {
+			if (systemInfo.contains("=")) {
+				String[] metaData = systemInfo.split("=");
+
+				if (metaData.length != 2) {
+					throw new IllegalArgumentException("System metadata is in invalid format!");
+				}
+				String key = metaData[0];
+				String vaule = metaData[1];
+				if (system.getMetadata() != null && system.getMetadata().containsKey(key)
+						&& system.getMetadata().get(key).equals(vaule)) {
+					return system.getId();
+				}
+			} else if (system.getSystemName().equals(systemInfo)) {
+				return system.getId();
+			}
+		}
+		throw new Exception("Could not find id for system with info: " + systemInfo);
+	}
+
+	// -------------------------------------------------------------------------------------------------
+	private long interfaceNameToId(final String interfaceName) throws Exception {
+		Assert.notNull(this.interfaces, "The list of possible interfaces is empty!");
+		for (ServiceInterfaceResponseDTO interfaceElement : this.interfaces) {
+			if (interfaceElement.getInterfaceName().equals(interfaceName)) {
+				return interfaceElement.getId();
+			}
+		}
+		throw new Exception("Could not find id for interface with name: " + interfaceName);
+	}
+
+	// -------------------------------------------------------------------------------------------------
+	private long serviceDefinitionToId(final String serviceDefinition) throws Exception {
+		Assert.notNull(this.services, "The ist of possible services is empty!");
+		for (ServiceDefinitionResponseDTO service : this.services) {
+			if (service.getServiceDefinition().equals(serviceDefinition)) {
+				return service.getId();
+			}
+		}
+		throw new Exception("Could not find id for service definition with name: " + serviceDefinition);
+	}
+
+	// -------------------------------------------------------------------------------------------------
+	private List<Integer> getRuleIdsToDelete(final List<AuthorizationIntraCloudResponseDTO> rules, final List<Integer> systemIds) {
+		List<Integer> result = new ArrayList<>();
+		for (AuthorizationIntraCloudResponseDTO rule : rules) {
+			for (int id : systemIds) {
+				if (rule.getConsumerSystem().getId() == id) {
+					result.add((int) rule.getId());
+				}
+			}
+		}
+		return result;
+	}
+
+	// -------------------------------------------------------------------------------------------------
+	private List<SystemResponseDTO> getSystems() {
+		logger.info("Get systems request started...");
+		SystemListResponseDTO response = arrowheadService.consumeServiceHTTP(SystemListResponseDTO.class,
+				HttpMethod.GET,
+				Utilities.createURI(getScheme(),
+						CommonConstants.$SERVICEREGISTRY_ADDRESS_WD.split(":")[1].replace("}", ""),
```

Comment:
This is really strange and only works if your Service Registry is on the localhost. What you need is the value of the @Value(CommonConstants.$SERVICEREGISTRY_ADDRESS_WD). (This should be a member)